cpp/tests/strings/slice_tests.cpp

```diff
@@ -268,6 +268,23 @@ TEST_F(StringsSliceTest, MaxPositions)
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(*results, expected);
 }
 
+TEST_F(StringsSliceTest, MultiByteChars)
+{
+  auto input = cudf::test::strings_column_wrapper({
+    // clang-format off
+    "quick brown fox jumped over the lazy brown dog; the fat cats jump in place without moving "
+    "the following code snippet demonstrates how to use search for values in an ordered range  "
+    "it returns the last position where value could be inserted without the ééééé ordering ",
+    "algorithms execution is parallelized as determined by an execution policy; this is a 12345"
+    "continuation of previous row to make sure string boundaries are honored 012345678901234567"
+    "01234567890é34567890012345678901234567890"
```

Comment:
why do we need this string to repro the issue? 
My guess: the issue is with `ééééé` and the rest of the string is there only to get to the size that triggers the execution policy that has the issue.

Reply:
Yes. I can add some comments explaining this. 
The position of the `é` characters and the length of the strings verify logic fix in this PR.

Reply:
A comment would be helpful, thank you!